### PR TITLE
fix regex & omit vars when no value

### LINKED ISSUE #1798: the problem of empty vars expressions in "Advanced Routing Matching Conditions"
# Feature request

## Please describe your feature

When adding a route, Dashboard will add an empty expression of `"vars":{}` by default:
image

However, only in the APISIX 2.5 version, APISIX will perform schema verification on the `"vars":{}` expression in the route. If there is `"vars":{}` in the route, the following error message will be reported:

```
2021/04/14 14:09:03 [error] 42#42: 1 [lua] config_etcd.lua:230: load_full_data(): failed to check item data of [/apisix/routes] err:failed to validate the 'vars' expression: rule too short ,val: {"uris":["/ecom/"],"vars":{},"hosts":["dev.url","develop.url"],"id":"350417588708180914","update_time":1618401644,"status":1,"create_time":1618394758,"name":"ecom_dev_v2","upstream":{"hash_on":"vars","scheme":"http","type":"roundrobin","timeout":{"read":6000,"connect":6000,"send":6000},"pass_host":"pass","nodes":[{"host":"ecom-edg.development.svc.cluster.local","weight":1,"port":8085,"priority":0}]},"priority":0,"methods":["GET","POST","PUT","DELETE"]}, context: init_worker_by_lua*
```

I see that the community Dashboard 2.6 is ready to be released, which will be used in conjunction with the APISIX 2.5 version. But I am not sure whether Dashboard 2.6 will add the empty expression of `"vars":{}` by default when adding routes. If there is, then maybe Dashboard 2.6 and APISIX 2.5 are used together, and the above-mentioned incompatibility problem may occur. 

Back-end related issues: https://github.com/apache/apisix/issues/4048

## Describe the solution you'd like

If Dashboard version 2.6 adds the empty expression of `"vars":{}` by default when adding routes, then we should remove it.

### PR DESCRIPTION
**Pull request by juzhiyuan**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

- [x] update Regex according to schema_def
- [x] omit vars field in Route when no value contained

**Related issues**

fix #1798 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: web/src/pages/Route/locales/en-US.ts [hunk 1]
```diff
@@ -75,10 +75,9 @@ export default {
   'page.route.form.itemExtraMessage.domain':
     'Domain Name or IP, support for generic Domain Name, for example: *.test.com',
   'page.route.form.itemRulesPatternMessage.domain':
-    'Only letters, numbers and * are supported. * can only be at the beginning, and only single * is supported',
+    'Only letters, numbers and * are supported, but * needs to be at the beginning.',
```
Comment (liuxiran) [inline web/src/pages/Route/locales/en-US.ts:78]:
```suggestion
    'Only letters, numbers, -,_ and * are supported, but * needs to be at the beginning.',
```

File: web/src/pages/Route/locales/zh-CN.ts [hunk 2]
```diff
@@ -74,10 +74,9 @@ export default {
   'page.route.form.itemLabel.redirectURI': '重定向路径',
   'page.route.form.itemExtraMessage.domain': '路由匹配的域名列表。支持泛域名，如：*.test.com',
   'page.route.form.itemRulesPatternMessage.domain':
-    '仅支持字母、数字和 * ，且 * 只能是在开头，支持单个 * ',
+    '仅支持字母、数字和 *，但 * 需要在开头位置。',
```
Comment (liuxiran) [inline web/src/pages/Route/locales/zh-CN.ts:77]:
```suggestion
    '仅支持字母、数字、-、_和 *，但 * 需要在开头位置。',
```

**Review by liuxiran — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Review by starsz — APPROVED** [review]
Regex LGTM.

**Review by moonming — CHANGES_REQUESTED** [review]
need test case

**Review by guoqqqi — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Comment by juzhiyuan** [comment]
ping @moonming to review.

**Comment by Firstsawyou** [comment]
cc @juzhiyuan   The conflict needs to be resolved.
Some users have repeatedly reported the bug of `regex`, we should fix it as soon as possible.

**Comment by iamayushdas** [comment]
I think i should be looking into this, as he is kinda busy i guess

**Comment by juzhiyuan** [comment]
> cc @juzhiyuan The conflict needs to be resolved.
> Some users have repeatedly reported the bug of `regex`, we should fix it as soon as possible.

ok, will fix today.

**Comment by iamayushdas** [comment]
> > cc @juzhiyuan The conflict needs to be resolved.
> > Some users have repeatedly reported the bug of `regex`, we should fix it as soon as possible.
> 
> ok, will fix today.

okay sir

**Comment by juzhiyuan** [comment]
duplicate with https://github.com/apache/apisix-dashboard/pull/1921

**Pull request closed without merging**